compiler/src/iree/compiler/Codegen/Dialect/VectorExt/IR/VectorExtAttrs.td

```diff
@@ -242,6 +242,12 @@ def NestedLayoutAttr : IREEVectorExt_Attr<"NestedLayout",
 
     // Get the undistributed shape that is subgroup x batch x outer x thread x element
     SmallVector<int64_t> getUndistributedPackedShape() const;
+
+    // Get the subgroup x batch x outer x thread x element shape for original dim
+    SmallVector<int64_t> getPackedShapeForUndistributedDim(int64_t dim) const;
+
+    // Get the distributed shape but has the same rank as the undistributed shape.
+    SmallVector<int64_t> getDistributedUnpackedShape() const;
```

Comment:
Echoing @bjacob here: https://github.com/iree-org/iree/pull/19905#discussion_r1942149508

These could be freestanding functions instead of class members if it's not too much trouble.

Reply:
Yes they could be (and I m happy to change to that).. but I d like to understand/learn the rationale especially since it uses the state of object to perform the computation required especially when they are MLIR attributes.

To me f(object, ... ) vs object.f(...) is traditional argument in C++ where the former is preferred for encapsulation reasons as it cannot access private/protected members but I thought that does not necessarily hold of MLIR tablegen'd attributes.